src/test/java/htsjdk/samtools/SamReaderFactoryTest.java

```diff
@@ -284,7 +284,37 @@ public void queryInputResourcePermutation(final SamInputResource resource) throw
         }
         reader.close();
     }
-    
+
+    public class NeverFilePathInpurResource extends PathInputResource {
+        public NeverFilePathInpurResource(Path pathResource) {
+            super(pathResource);
+        }
+
+        @Override
+        public File asFile() {
+            return null;
+        }
+    }
+
+    @Test
```

Comment:
The query test actually doesn't fail if it can't use the index, it just falls back to not using the index and emits a warning.  That seemed strange (to log a warning in a test rathr than fail) but I didn't want to change it's behavior as it predates me.  So I added the second test instead to assert the index is available which combined with the first test ensure it's both available AND does the right thing in query

Techically, the Then I patched SamFileReader